src/main/kotlin/util/FSTSerializer.kt

```diff
@@ -52,8 +61,43 @@ internal class StreamDecoderFactory(
     }
 }
 
-internal class FSTCustomSteamDecoder(conf: FSTConfiguration) : FSTStreamDecoder(conf) {
-    override fun readStringUTF(): String {
-        return super.readStringUTF().weakIntern()
+class FSTCustomConfiguration : FSTConfiguration(null) {
+
+    init {
+        initDefaultFstConfigurationInternal(this)
+    }
+
+    override fun encodeToStream(out: OutputStream, toSerialize: Any) {
+        val objectOutput = objectOutput;
+        objectOutput.writeObject(toSerialize)
+        val written = objectOutput.written
+        out.write(written ushr 0 and 0xFF)
+        out.write(written ushr 8 and 0xFF)
+        out.write(written ushr 16 and 0xFF)
+        out.write(written ushr 24 and 0xFF)
+        out.write(objectOutput.buffer, 0, written)
+        objectOutput.flush()
+        objectOutput.resetForReUse(byteArrayOf())
```

Comment:
why doesn't extend this method by this? :
```suggestion
        super.encodeToStream(out, toSerialize)
        objectOutput.resetForReUse(byteArrayOf())
```